src/server/JasmineGraphServer.cpp

```diff
@@ -2280,6 +2347,269 @@ bool JasmineGraphServer::initiateClient(std::string host, int port, int dataPort
+    serv_addr.sin_family = AF_INET;
+    bcopy((char *) server->h_addr,
+          (char *) &serv_addr.sin_addr.s_addr,
+          server->h_length);
+    serv_addr.sin_port = htons(port);
+    if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+        std::cerr << "ERROR connecting" << std::endl;
+        //TODO::exit
+    }
+
+    bzero(data, FED_DATA_LENGTH);
+    write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
+    server_logger.log("Sent fed : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
+    bzero(data, FED_DATA_LENGTH);
+    read(sockfd, data, FED_DATA_LENGTH);
+    string response = (data);
+
+    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
+        server_logger.log("Received fed : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
+        string server_host = utils.getJasmineGraphProperty("org.jasminegraph.server.host");
+        write(sockfd, server_host.c_str(), server_host.size());
+        server_logger.log("Sent fed : " + server_host, "info");
+
+        write(sockfd, JasmineGraphInstanceProtocol::INITIATE_ORG_SERVER.c_str(),
+              JasmineGraphInstanceProtocol::INITIATE_ORG_SERVER.size());
+        server_logger.log("Sent fed : " + JasmineGraphInstanceProtocol::INITIATE_ORG_SERVER, "info");
+        bzero(data, FED_DATA_LENGTH);
+        read(sockfd, data, FED_DATA_LENGTH);
+        response = (data);
+        response = utils.trim_copy(response, " \f\n\r\t\v");
+        std::cout << response << std::endl;
+        if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
+            server_logger.log("Received fed : " + JasmineGraphInstanceProtocol::OK, "info");
+            write(sockfd, (trainingArgs).c_str(), (trainingArgs).size());
+            server_logger.log("Sent fed : training args " + trainingArgs, "info");
+            bzero(data, FED_DATA_LENGTH);
+            return 0;
+            }
+    } else {
+        server_logger.log("There was an error in the invoking training process and the response is :: " + response,
+                        "error");
+    }
+
+    close(sockfd);
+    return 0;
+}
+
+bool JasmineGraphServer::receiveGlobalWeights(std::string host, int port, std::string trainingArgs, int iteration, std::string partCount) {
+    
+    int HEADER_LENGTH = 10;
+    Utils utils;
+    bool result = true;
+    int sockfd;
+    
+    bool loop = false;
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
+
+    std::string weights_file = utils.getJasmineGraphProperty("org.jasminegraph.fl.weights.file");
+    std::string flagPath = utils.getJasmineGraphProperty("org.jasminegraph.fl.flag.file");
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+    if (sockfd < 0) {
+        std::cerr << "Cannot accept connection" << std::endl;
```

Comment:
Use logging framework

Reply:
Fixed in 3838b63e0c10352eee23ceff37cafc3492b3c14e

Reply:
Done